library/src/com/squareup/timessquare/CalendarPickerView.java

```diff
@@ -644,7 +663,16 @@ public void highlightDates(Collection<Date> dates) {
     }
 
     adapter.notifyDataSetChanged();
-    setAdapter(adapter);
```

Comment:
ah!  yeah, let's just replace both lines with the `validateAndUpdate()` call.  good catch!